code/modules/paperwork/photocopier.dm

```diff
@@ -410,6 +410,8 @@
 				. = TRUE
 		if("scandocument")
 			scan_document()
+		if("ai_text")
+			ai_text()
```

Comment:
have this pass `ui.user`, you'll need to modify `ui_act()`'s parameters to include the TGUI datum

Reply:
this should do
https://github.com/ParadiseSS13/Paradise/pull/22653/commits/a1e324101d6ce0d5f8a8c000aefc855b028b7a8e